### PR TITLE
use type info to assign an affinity thread

### PR DESCRIPTION
**Pull request by duke8253**

### CODE REVIEW

**Comment by SolidWallOfCode** [comment]
I think it should be `assign_affinity_by_type` instead of `assign_thread_as_affinity`. I meaning, assigning a thread is affinity, that name doesn't make much sense.

File: iocore/eventsystem/P_UnixEventProcessor.h [hunk 1]
```diff
@@ -61,6 +61,19 @@ EventProcessor::assign_thread(EventType etype)
...
 }
 
+// Returns the thread affinity if it is present in the continuation.
+// Otherwise assigns the next available thread of "etype" to be the affinity thread and return it.
+TS_INLINE EThread *
+EventProcessor::assign_thread_as_affinity(Continuation *cont, EventType etype)
+{
+  EThread *ethread = cont->getThreadAffinity();
+  if (ethread == nullptr) {
+    ethread = assign_thread(etype);
```
[the remark this answers is not in the document]
Comment (duke8253) [reply iocore/eventsystem/P_UnixEventProcessor.h]: But I need to return `ethread` afterwards.

File: iocore/eventsystem/P_UnixEventProcessor.h [hunk 2]
```diff
@@ -61,6 +61,19 @@ EventProcessor::assign_thread(EventType etype)
   return tg->_thread[next];
 }
 
+// Returns the thread affinity if it is present in the continuation.
+// Otherwise assigns the next available thread of "etype" to be the affinity thread and return it.
+TS_INLINE EThread *
```
[the remark this answers is not in the document]
Comment (duke8253) [reply iocore/eventsystem/P_UnixEventProcessor.h]: Should I change all the other `TS_INLINE` to `inline` in this file?

[the remark this answers is not in the document]
Comment (SolidWallOfCode) [reply iocore/eventsystem/P_UnixEventProcessor.h]: Yes.

**Pull request closed without merging**